testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterWithUserProfileTest.java

```diff
@@ -510,14 +507,19 @@ private void assertUserRegistered(String userId, String username, String email,
         Assert.assertNotNull(user);
         Assert.assertNotNull(user.getCreatedTimestamp());
         // test that timestamp is current with 10s tollerance
-        Assert.assertTrue((System.currentTimeMillis() - user.getCreatedTimestamp()) < 10000);
+        assertTrue((System.currentTimeMillis() - user.getCreatedTimestamp()) < 10000);
         // test user info is set from form
         assertEquals(username.toLowerCase(), user.getUsername());
         assertEquals(email.toLowerCase(), user.getEmail());
         assertEquals(firstName, user.getFirstName());
-        assertEquals(lastName, user.getLastName());
+
+        if (StringUtils.isEmpty(lastName)) {
+            assertTrue(StringUtils.isEmpty(user.getLastName()));
+        } else {
+            assertEquals(lastName, user.getLastName());
```

Comment:
Nitpick: Please, always use `org.hamcrest.MatcherAssert.assertThat` instead of `Assert.assertEquals`. For this PR, it's used only here, so you can leave it as it is. It's just advice to the future.